mitosheet/src/mito/components/SearchBar.tsx

```diff
@@ -189,56 +191,89 @@ export const SearchBar: React.FC<SearchBarProps> = (props) => {
+                            handleCurrentMatchChange('next');
+                        }
+                    }}
+                    onKeyUp={(e: React.KeyboardEvent) => {
+                        // onKeyDown can't detect shift+enter, so we use onKeyUp to detect it
+                        if (e.key === 'Enter' && e.shiftKey) {
+                            handleCurrentMatchChange('prev');
+                        }
+                    }}
+                    className={classNames('mito-input')}
+                    placeholder='Find...'
+                    autoFocus
+                />
+                <span>{finalMatchInfo}</span>
+                {/* This button jumps to the previous match */}
+                <button
+                    className='mito-search-button'
+                    onClick={() => {
+                        handleCurrentMatchChange('prev')
+                    }}
+                >
+                    <SearchNavigateIcon width='17' height='17' direction='up' strokeColor='var(--mito-text)' strokeWidth={1} />
+                </button>
+                {/* This button jumps to the next match */}
+                <button
+                    className='mito-search-button'
+                    onClick={() => {
+                        handleCurrentMatchChange('next')
+                    }}
+                >
+                    <SearchNavigateIcon width='17' height='17' direction='down' strokeColor='var(--mito-text)' strokeWidth={1} />   
+                </button>
+                {/* This button closes the search bar. */}
+                <button
+                    className='mito-search-button'
+                    onClick={() => {
+                        setUIState({
+                            ...uiState,
+                            currOpenSearch: { isOpen: false, currentMatchIndex: 0, matches: [] }
+                        })
+                    }}
+                >
+                    <XIcon strokeWidth='1' width='15' height='15' />
+                </button>
+            </div>
+            {isExpanded && <div className='mito-search-bar-replace'>
+                <Input
+                    value={replaceValue}
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                        setReplaceValue(e.target.value);
+                    }}
+                    onKeyDown={(e: React.KeyboardEvent) => {
+                        if (e.key === 'Enter') {
+                            void mitoAPI.editReplace(uiState.selectedSheetIndex, searchValue ?? '', replaceValue ?? '')
+                        }
+                    }}
+                    className='mito-input'
+                    placeholder='Replace...'
+                    autoFocus
+                />
+                <button className='mito-search-button' onClick={() => {
+                    void mitoAPI.editReplace(uiState.selectedSheetIndex, searchValue ?? '', replaceValue ?? '')
```

Comment:
Should we maybe bail if they aren't defined?

Reply:
I think if the user wants to use empty string / undefined as the replacement value that should be allowed - I use this sometimes to remove strings from larger strings